dix/dix_priv.h

```diff
@@ -689,4 +690,18 @@ static inline ClientPtr dixLookupXIDOwner(XID xid)
     return NullClient;
 }
 
+/*
+ * @brief write rpc buffer to client and then clear it
+ *
+ * @param pClient the client to write buffer to
+ * @param rpcbuf  the buffer whose contents will be written
+ * @return the result of WriteToClient() call
+ */
+static inline int WriteRpcbufToClient(ClientPtr pClient,
```

Comment:
It all looks good now, but im wondering if you can put an assert or something here that checks that rpcbuf->error isn't set. I know that WriteToClient wont write anything to the client in that case since rpcbuf->buffer will be NULL, but it will be hard to debug if it happens since there wont really be any error from the client, it will be stuck waiting for the reply of length as specified in the reply header.